opta/core/cloud_provider.py

```diff
@@ -0,0 +1,52 @@
+import boto3
+from botocore.config import Config
+from botocore.exceptions import ClientError
+
+from opta.core.terraform2 import TerraformFile
+from opta.layer2 import Layer
+
+
+class CloudProvider:
```

Comment:
this is an opportunity to document a bit, do you mind adding a one liner for each cloud function to implement?